app/Http/Controllers/RegulationController.php

```diff
@@ -22,14 +30,132 @@ public function getAllSources(){
     public function getAllDestinations() {
         return response()->json(ReuseNode::destinations());
     }
+    public function getAllAllowed() {
+        return response()->json(Allowed::all());
+    }
+
     public function addRegulationSubmit(Request $request) {
         if (empty($request->newRegList))
             return redirect()->route('regSubmit')->with(['alert' => 'danger', 'alertMessage' => 'You submitted an empty regulation']);
 
-        // $source = new Source();
-        // $source->sourceName = $request->source;
-        // $source->save();
+        $regLists = json_decode($request->newRegList, true);
+        $regArea = "";
+
+        //There is only a State
+        if($regLists[0]['$county'] == '' || $regLists[0]['$county'] == 'Choose...')
+        {
+            foreach($regLists as $regList)
+            {
+
+                $codesLink = new Links();
+                $codesLink->linkText = $regList['$codesLink'];
+                $codesLink->save();
+
+                $permitLink = new Links();
+                $permitLink->linkText = $regList['$permitLink'];
+                $permitLink->save();
+
+                $insentivesLink = new Links();
+                $insentivesLink->linkText = $regList['$insentivesLink'];
+                $insentivesLink->save();
+
+                $moreInfoLink = new Links();
+                $moreInfoLink->linkText = $regList['$moreInfoLink'];
+                $moreInfoLink->save();
```

Comment:
For the links, could we not create new entries if the link field is empty? I believe the PHP function `empty($variable)` can tell if a variable has no meaningful content. Currently, just from testing this, we have ~125 entries in the Links table, and the vast majority of them have empty content.

Reply:
Also, you don't have to do this now, but I might refactor this code to create stuff outside of these loops - perhaps a `createEntry()` function. Passing variables by reference with PHP is a thing (using good old `&`), so that might be one way to do it. Again, feel free to leave this part for a future story.

Reply:
I had a feeling this one would come up. I thought it was required for the merge table to have a link to all four of the links. If this isn't the case, I won't create the links if they are empty. If so, maybe we just shouldn't display empty links? And why do we need a separate function to create things outside of the loop? If it would effect the system poorly in some way, I'm totally down to change. If it's just a different way of doing the same thing, I'll probably just leave it as is.